snow/engine/common/bootstrapper.go

```diff
@@ -179,12 +257,20 @@ func (b *Bootstrapper) Accepted(validatorID ids.ShortID, requestID uint32, conta
 		}
 	}
 
-	if size := len(accepted); size == 0 && b.Beacons.Len() > 0 {
-		b.Ctx.Log.Info("Bootstrapping finished with no accepted frontier. This is likely a result of failing to be able to connect to the specified bootstraps, or no transactions have been issued on this chain yet")
-	} else {
-		b.Ctx.Log.Info("Bootstrapping started syncing with %d vertices in the accepted frontier", size)
+	// if we don't have enough weight for the bootstrap to be accepted then restart or fail the bootstrap
+	size := len(accepted)
+
+	if size == 0 && b.Beacons.Len() > 0 {
+		b.Ctx.Log.Info("Bootstrapping finished with no accepted frontier. This is likely a result of failing to "+
+			"be able to connect to the specified bootstraps, or no transactions have been issued on this chain yet"+
+			" - Number of beacons: %d - bootstrap attempt: %d", b.Beacons.Len(), b.bootstrapAttempts)
+		if b.Config.Ctx.RetryBootstrap {
+			return b.RestartBootstrap()
+		}
 	}
 
+	b.Ctx.Log.Info("Bootstrapping started syncing with %d vertices in the accepted frontier", size)
```

Comment:
lets reset `b.bootstrapAttempts` to 0 here.

Reply:
Took a slightly different approach - thoughts?

```go
func (b *Bootstrapper) RestartBootstrap(reset bool) error {

	// resets the attempts when we're pulling blocks/vertices
	// we don't want to fail the bootstrap at that stage
	if reset {
		b.bootstrapAttempts = 0
	}

	if b.bootstrapAttempts >= 50 {
		return fmt.Errorf("failed to boostrap the chain after %d attempts", b.bootstrapAttempts)
	}
```